realsense2_camera/src/base_realsense_node.cpp

```diff
@@ -1048,6 +1081,151 @@ void BaseRealSenseNode::publishStaticTransforms()
     }
 }
 
+/* Modified for Brain */
+static void output_unit_vector(unsigned int height, unsigned int width, float unitVectors[FRAME_MAX_HEIGHT][FRAME_MAX_WIDTH * 3])
+{
+	unsigned int i, j;
+	FILE *fp; /* output to .obj file with a certain format to be shown by some tool */
+
+	std::cerr << "\nUnit Test for Realsense Unit Vector:\nOutput the saved data: frame width is " \
+		<< width << " and frame height is " << height << " \n\n" << std::endl;
+
+	std::cerr << "*** Beginning of unit vectors ***\n" << std::endl;
+
+	fp = fopen("./realsense_unit_vectors.obj","w");
+	if (!fp) {
+		std::cerr << "\nFailed to open a file for realsense_unit_vectors.obj!\n\n" << std::endl;
+		return;
+	}
+
+	std::cerr << "\nOutput unit_vectors to ./realsense_unit_vectors.obj!\n\n" << std::endl;
+
+	for (i = 0; i < height; i++)
+		for (j = 0; j < width * 3; j = j + 3)
+			fprintf(fp, "v %.6f %.6f %.6f\n", unitVectors[i][j], unitVectors[i][j + 1], unitVectors[i][j + 2]);
+
+	fclose(fp);
+
+	std::cerr << "\n*** End of unit vectors ***\n\n" << std::endl;
+}
+
+void BaseRealSenseNode::publishUnitVectorsTopic(const ros::Time& t, const std::map<stream_index_pair, bool>& is_frame_arrived)
+{
+	static float num_valid;
+	static bool unit_vector_mature;
+	static unsigned int height = 240, width = 424, sequence;
```

Comment:
It doesn't look like _height_ or _width_ need to be defined here, since they're getting reassigned a few lines down